xclip.c

```diff
@@ -416,7 +417,8 @@ doIn(Window win, const char *progname)
     if ((fil_number == 0) && ffilt) {
 	fwrite(sel_buf, sizeof(char), sel_len, stdout);
 	fclose(stdout);
-    }
+    } else
+	free(fil_names);
```

Comment:
It is not clear to me here why the free is in the else section of this if (fil_number == 0) && ffilt. Even if that is correct, it is confusing and would require a comment. Why not just add a new if statement that checks if (fil_names != NULL)? Also, are we *absolutely* sure fil_names won't be used after this? If so, we should set fil_names = NULL.